frontend/eslint-local-rules/matchingTranslationKeys.js

```diff
@@ -0,0 +1,214 @@
+const path = require('path')
+// TODO move this file to common and get this require to still work, in frontend and print
+const utils = require('eslint-plugin-vue/lib/utils/index.js')
+
+/**
+ * List of components with special props which accept translation keys.
+ * Key is the component name (in kebab-case, as used in the template), and value is an array
+ * of props which expect a translation key as value.
+ * E.g. <dialog-form submit-label="global.button.submit" :cancel-label="'global.button.cancel'"></dialog-form>
+ */
+const translationKeyComponentProps = {
```

Comment:
Can we generalize this instead of whitelisting all usages? E.g. all properties that start with `i18n`?

Also (not related to this PR): I'm not sure passing keys is really needed for DialogForm. We might want to refactor this.